src/overlays/actors/ovl_Obj_Mure/z_obj_mure.c

```diff
@@ -26,43 +40,375 @@ const ActorInit Obj_Mure_InitVars = {
+            break;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Mure/ObjMure_Init.s")
+void ObjMure_KillActorsImpl(ObjMure* this, GlobalContext* globalCtx) {
+    s32 maxChildren = ObjMure_GetMaxChildSpawns(this);
+    s32 i;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Mure/ObjMure_Destroy.s")
+    for (i = 0; i < maxChildren; i++) {
+        switch (this->actorSpawnUnk[i]) {
+            case 1:
+                this->actorSpawnPtrList[i] = NULL;
+                break;
+            case 2:
+                if (this->actorSpawnPtrList[i] != NULL) {
+                    Actor_Kill(this->actorSpawnPtrList[i]);
+                    this->actorSpawnPtrList[i] = NULL;
+                }
+                break;
+            default:
+                if (this->actorSpawnPtrList[i] != NULL) {
+                    if (Actor_HasParent(this->actorSpawnPtrList[i], globalCtx)) {
+                        this->actorSpawnPtrList[i] = NULL;
+                    } else {
+                        Actor_Kill(this->actorSpawnPtrList[i]);
+                        this->actorSpawnPtrList[i] = NULL;
+                    }
+                }
+                break;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Mure/func_80B98C88.s")
+void ObjMure_KillActors(ObjMure* this, GlobalContext* globalCtx) {
+    ObjMure_KillActorsImpl(this, globalCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Mure/func_80B98CB4.s")
+void func_80B99224(ObjMure* this, GlobalContext* globalCtx) {
+    s32 maxChildren = ObjMure_GetMaxChildSpawns(this);
+    s32 i;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Mure/func_80B98D18.s")
+    for (i = 0; i < maxChildren; i++) {
+        if (this->actorSpawnPtrList[i] != NULL) {
+            if (this->actorSpawnUnk[i] == 0) {
+                if (this->actorSpawnPtrList[i]->update != NULL) {
+                    if (this->actorSpawnPtrList[i]->flags & 0x800) {
+                        this->actorSpawnUnk[i] = 2;
+                    }
+                } else {
+                    this->actorSpawnUnk[i] = 1;
+                    this->actorSpawnPtrList[i] = NULL;
+                }
+            } else if ((this->actorSpawnUnk[i] == 2) && (this->actorSpawnPtrList[i]->update == NULL)) {
+                this->actorSpawnUnk[i] = 1;
+                this->actorSpawnPtrList[i] = NULL;
+            }
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Mure/func_80B98F38.s")
+void ObjMure_InitialAction(ObjMure* this, GlobalContext* globalCtx) {
+    this->actionFunc = ObjMure_IdleOffCam;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Mure/func_80B990BC.s")
+void ObjMure_IdleOffCam(ObjMure* this, GlobalContext* globalCtx) {
+    if (fabsf(this->actor.projectedPos.z) < sZClip[this->type]) {
+        this->actionFunc = ObjMure_IdleOnCam;
```

Comment:
```suggestion
void ObjMure_IdleCulled(ObjMure* this, GlobalContext* globalCtx) {
    if (fabsf(this->actor.projectedPos.z) < sZClip[this->type]) {
        this->actionFunc = ObjMure_IdleUnculled;
```
And I wouldn't call them Idle, I know fishs/bugs aren't doing much but those two Idle_ functions are the only action functions the actor uses, can't tell it's idleing 100% of the time 😅 
Maybe just drop that Idle prefix? I don't see what could be put in its place